### PR TITLE
feature: add landing page

### PR DESCRIPTION
**Pull request by 0xsubi**

### CODE REVIEW

File: helper.go [hunk 1]
```diff
@@ -146,3 +138,10 @@ func getRequestURL(w http.ResponseWriter, r *http.Request) *url.URL {
 
 	return URL
 }
+
+func serveLandingPage(w http.ResponseWriter, r *http.Request)  {
+	b, _ := ioutil.ReadFile("public/index.html")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write(b)
+	return
```
Comment (Shivam010) [inline helper.go]: Set Content-type header to `text/html`
```suggestion
	w.Header().Set("Content-Type", "text/html; charset=utf-8")
	return
```

File: main.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ func (*handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	URL := getRequestURL(w, r)
 	if URL == nil { // invalid URL
+		serveLandingPage(w, r)
```
Comment (Shivam010) [inline main.go]: Do not call the landing page here...
As `URL` can be `nil` in some other case also, in such cases the `Return` method errors are overwritten.
```suggestion
```

File: helper.go [hunk 3]
```diff
@@ -113,15 +114,6 @@ func addHeaders(w http.ResponseWriter, r *http.Request) bool {
...
 	if r.URL.Path == "" || r.URL.Path == "/" {
-		fmt.Println("Root Request")
-		Return(w, &Error{
-			Code:    http.StatusPreconditionFailed,
-			Message: "URL not provided",
-			Detail: map[string]interface{}{
-				"method":       r.Method,
-				"requestedURL": r.URL.String(),
-			},
-		})
```
Comment (Shivam010) [inline helper.go:139]: Call landing page here...

File: public/index.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    <p id="description">Proxy your requests just by prefixing your request URL with: https://non-cors.herokuapp.com/ with all the headers and payload as it is..</p>
+    <br>
+    <br>
+
+    <b>Example: </b>
+    <p id="example">GET https://joke-api-strict-cors.appspot.com/jokes/ten    ==>    GET https://non-cors.herokuapp.com/https://joke-api-strict-cors.appspot.com/jokes/ten</p>
+    <a href="https://github.com/Shivam010/bypass-cors" target="_blank"><img id="github_icon" src="https://camo.githubusercontent.com/d0518022b7a02d405ad5112a0c8aa455cbfe952e/68747470733a2f2f6564656e742e6769746875622e696f2f537570657254696e7949636f6e732f696d616765732f7376672f6769746875622e737667">Github Repo</a>
+
+</div>
```
Comment (Shivam010) [inline public/index.html:45]: Try to use all the data from Readme file in the Landing page

**Review by Shivam010 — CHANGES_REQUESTED** [review]
Hi @sudhabindu1 
I have reviewed your changes & requested some changes.

**NOTE:** Also, change the target branch to `dev`, merge and incorporate its changes and then, submit it, please

**Comment by 0xsubi** [comment]
@Shivam010 I have made the changes as suggested. But not sure on the bit about using all the data from readme.md. I think it would make the homepage cluttered. And we're linking to the repo anyway.

**Pull request closed without merging**